services/web/server/src/simcore_service_webserver/api_keys/_api.py

```diff
@@ -0,0 +1,73 @@
+import logging
+from typing import Final
+
+from aiohttp import web
+from models_library.api_schemas_webserver.auth import ApiKeyCreate, ApiKeyGet
+from models_library.products import ProductName
+from models_library.users import UserID
+
+from ..login.utils import get_random_string
+from ._db import ApiKeyRepo
+
+_logger = logging.getLogger(__name__)
+
+_KEY_LEN: Final = 10
+_SECRET_LEN: Final = 30
+
+
+async def list_api_keys(
+    app: web.Application,
+    *,
+    user_id: UserID,
+    product_name: ProductName,
+) -> list[str]:
+    repo = ApiKeyRepo.create_from_app(app)
+    names: list[str] = await repo.list_names(user_id=user_id, product_name=product_name)
+    return names
```

Comment:
Only listing. I don't have an endpoint to get an API key by name. I would like to search for one and if it's not present I would create it. This would be the workflow I would desire.

Reply:
Don't change this I will add it when required.